packages/connector-aliyun-sms/src/index.ts

```diff
@@ -38,32 +39,75 @@ export default class AliyunSmsConnector implements SmsConnector {
       new ConnectorError(ConnectorErrorCodes.TemplateNotFound, `Cannot find template!`)
     );
 
-    const httpResponse = await sendSms(
-      {
-        AccessKeyId: accessKeyId,
-        PhoneNumbers: phone,
-        SignName: signName,
-        TemplateCode: template.code,
-        TemplateParam: JSON.stringify({ code }),
-      },
-      accessKeySecret
-    );
+    try {
+      const httpResponse = await sendSms(
+        {
+          AccessKeyId: accessKeyId,
+          PhoneNumbers: phone,
+          SignName: signName,
+          TemplateCode: template.code,
+          TemplateParam: JSON.stringify({ code }),
+        },
+        accessKeySecret
+      );
 
-    const { body: rawBody } = httpResponse;
-    const result = sendSmsResponseGuard.safeParse(JSON.parse(rawBody));
+      const { body: rawBody } = httpResponse;
+      const result = sendSmsResponseGuard.safeParse(JSON.parse(rawBody));
 
-    if (!result.success) {
-      throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error.message);
-    }
+      if (!result.success) {
+        throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error.message);
+      }
+
+      const { Code } = result.data;
 
-    const { Code } = result.data;
+      /**
+       * You may wonder why the errorHandler is called twice here. We found that the SendSms API of
+       * the Aliyun SMS service may throw an error directly after receiving the request, or it may
+       * need developers to determine whether the request succeeded by `Code` getting from the response.
+       */
+      this.errorHandler(rawBody, Code);
```

Comment:
nit: maybe we'd better split `errorHandler` and deal with `Code` separately?

Reply:
as you can see, the server-side throws an error in two cases, returning a response that contains an error message and error description; another case that throws an error w/o giving a response.
The error handler was implemented to handle these two cases. Or we can implement two error handlers and one for each previously mentioned case.